### ISSUE TITLE
ReadTheDoc documentation failure

### ISSUE THREAD

**Issue by mattberjon** [earswideopen/ewo-core#18]
The actual documentation do not compile properly on readthedoc because of the installation of PyAudio on their server (that is far from being required...). It needs to be fixed.

```

Collecting PyAudio>=0.2.9 (from -r /home/docs/checkouts/readthedocs.org/user_builds/ewo-core/checkouts/latest/requirements.txt (line 1))
/home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading PyAudio-0.2.9.tar.gz (289kB)
Building wheels for collected packages: PyAudio
  Running setup.py bdist_wheel for PyAudio: started
  Running setup.py bdist_wheel for PyAudio: finished with status 'error'
  Complete output from command /home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-NXcjhc/PyAudio/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmpVi0b4gpip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-2.7
  copying src/pyaudio.py -> build/lib.linux-x86_64-2.7
  running build_ext
  building '_portaudio' extension
  creating build/temp.linux-x86_64-2.7
  creating build/temp.linux-x86_64-2.7/src
  x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c src/_portaudiomodule.c -o build/temp.linux-x86_64-2.7/src/_portaudiomodule.o
  src/_portaudiomodule.c:29:23: fatal error: portaudio.h: No such file or directory
   #include "portaudio.h"
                         ^
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for PyAudio
  Running setup.py clean for PyAudio
Failed to build PyAudio
Installing collected packages: PyAudio
  Running setup.py install for PyAudio: started
    Running setup.py install for PyAudio: finished with status 'error'
    Complete output from command /home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-NXcjhc/PyAudio/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-cHvKSQ-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/include/site/python2.7/PyAudio:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-2.7
    copying src/pyaudio.py -> build/lib.linux-x86_64-2.7
    running build_ext
    building '_portaudio' extension
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/src
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c src/_portaudiomodule.c -o build/temp.linux-x86_64-2.7/src/_portaudiomodule.o
    src/_portaudiomodule.c:29:23: fatal error: portaudio.h: No such file or directory
     #include "portaudio.h"
                           ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-NXcjhc/PyAudio/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-cHvKSQ-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/docs/checkouts/readthedocs.org/user_builds/ewo-core/envs/latest/include/site/python2.7/PyAudio" failed with error code 1 in /tmp/pip-build-NXcjhc/PyAudio/
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Command time: 2s Return: 1
```

**Comment by mattberjon** [comment]
It seems that the bug still exists. I reopen it.

**Comment by mattberjon** [comment]
This had been resolved thanks to the commit 9819c93. The help of the documentation from the [Astropy][l2] project as well as the official [ReadTheDocs documentation][l2] has been very helpful.

[l1]: http://docs.astropy.org/en/stable/development/affiliated-packages.html
[l2]: http://docs.readthedocs.io/en/latest/yaml-config.html
